### DISCUSSION TITLE
Why do you need your own go compiler + stdlib?

### DISCUSSION THREAD

**Discussion by luantak** [encoredev/encore#8] [Help] [answered]
I'm a little confused as to why you distribute an entire go toolchain with the standard libary.
Can you elaborate why you need this? 
I would feel more confident in this project, if it doesn't provide an entire separate go toolchain.
Why can't encore be installed via `go get/ go install`?

Even if you need to do this because you do some compiler magic etc. providing precompiled libaries under `encore-go/pkg/linux_amd64` seems a little excessive.

**Comment by eandre** [comment, marked as answer]
Thanks for asking. Encore's request semantics as well as automatic tracing require integration into the runtime. The changes are [all open source](https://github.com/encoredev/go) and constitute a very small amount of code, mostly involving adding some hooks. You are more than welcome to build the toolchain yourself.

To my knowledge the packaging process is the same as for the regular Go releases, and the precompiled libraries are present because not all environments have the tools installed to compile the standard library and cgo.
